test/python/visualization/test_gate_map.py

```diff
@@ -0,0 +1,109 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017, 2018.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""A test for visualizing device coupling maps"""
+import unittest
+import os
+import matplotlib.pyplot as plt
+from qiskit.test.mock import FakeProvider
+from qiskit.test import QiskitTestCase
+from qiskit.visualization.gate_map import _GraphDist, plot_gate_map
+from .visualization import path_to_diagram_reference, QiskitVisualizationTestCase
+
+
+class TestGateMap(QiskitVisualizationTestCase):
+    """ visual tests for plot_gate_map """
+    backends = []
+
+    def setUp(self):
+        """ setup for backend """
+        self.backends = list(filter(lambda x:
+                                    not (x.configuration().simulator
+                                         or x.configuration().n_qubits == 2),
+                                    FakeProvider().backends()))
+
+    def test_plot_gate_map(self):
+        """ tests plotting of gate map of a device (20 qubit, 16 qubit, 14 qubit and 5 qubit)"""
+        for i in self.backends:
```

Comment:
Could we make this name more descriptive? For example `i` -> `backend`

Reply:
Alright! have done this too. Thank you once again for reviewing my PR :)